lib/sda/processor.go

```diff
@@ -81,20 +82,47 @@ func (p *ServiceProcessor) ProcessClientRequest(e *network.Entity,
 // ProcessServiceMessage is to implement the Service interface.
 func (p *ServiceProcessor) ProcessServiceMessage(e *network.Entity,
 	s *ServiceMessage) {
-	cr := &ClientRequest{
-		Data: s.Data,
+	p.GetReply(e, s.Data)
+}
+
+// SendISM takes the message and sends it to the corresponding service
+func (p *ServiceProcessor) SendISM(e *network.Entity, msg network.ProtocolMessage) error {
+	sName := ServiceFactory.Name(p.Context.GetID())
+	sm, err := CreateServiceMessage(sName, msg)
+	if err != nil {
+		return err
 	}
-	p.GetReply(e, cr)
+	dbg.Lvl4("Raw-sending to", e)
+	return p.SendRaw(e, sm)
+}
+
+// SendISMOthers sends an InterServiceMessage to all other services
+func (p *ServiceProcessor) SendISMOthers(el *EntityList, msg network.ProtocolMessage) error {
+	errStrs := []string{}
```

Comment:
Nitpick: we should prefer `var errStrs []string` because it does not allocate anything if we never append on it. 
https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices